base.sh

```diff
@@ -0,0 +1,549 @@
+##
+# Copyright 2009-2014 Ghent University
+#
+# This file is part of csub,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/csub
+#
+# csub is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as
+# published by the Free Software Foundation, either version 2 of
+# the License, or (at your option) any later version.
+#
+# csub is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Library General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with csub. If not, see <http://www.gnu.org/licenses/>.
+#
+##
+
+
+##
+## this is the basic script that does it all
+## (no need for /bin/bash, is taken care of by BASEHEADER)
+##
+
+# empty job_pids cache
+job_pids_cache=""
+
+# unset %(CSUB_SERVER)s, can cause trouble when csub
+# submits to non-default cluster
+# e.g. qsub sets PBS_SERVER value to compiled-in default
+# which may cause next job to be submitted to wrong cluster
+unset %(CSUB_SERVER)s
+
+# make sure qsub is available
+module load jobs
+
+jobname=${%(CSUB_JOBNAME)s}
+## jobname_stripped is set in BASEHEADER
+scriptname="${jobname_stripped}.sh"
+localdir=$%(CSUB_SCRATCH_NODE)s/$jobname
+
+myecho () {
+    echo "$1"
+}
+
+if [ -z "$%(CSUB_SCRATCH_NODE)s" ]
+then
+    echo "%(CSUB_SCRATCH_NODE)s undefined"
+    env|sort|grep %(CSUB_ORG)s
+    exit 1
+fi
+
+%(prologue)s $%(CSUB_JOBID)s "" "" $%(CSUB_JOBNAME)s %(cleanup_chkpt)d
+if [ $? -ne 0 ]
+then
+	echo "ERROR! Prologue script abnormal exit."
+	exit 1
+fi
+
+if [ ! -d $localdir ]
+then
+    ## no result from prologue -> shared checkpoint dir
+    myecho "No localdir $localdir found (No local checkpoint)"
+
+    localdir=$%(CSUB_SCRATCH)s/chkpt/$jobname
+
+    # this directory should be created by csub
+    if [ ! -d $localdir ]
+    then
+    	## initial array job?
+    	localdir_initial=$%(CSUB_SCRATCH)s/chkpt/$jobname_stripped
+    	if [ ! -f $localdir/%(chkptsubdir)s/chkpt.count ] && [ -d $localdir_initial ]
+    	then
+    		# copy initial job directory for array jobs
+    		cp -r $localdir_initial $localdir
+    	else
+        	## problem
+        	myecho "No localdir $localdir found (No shared checkpoint)"
+        	exit 1
+        fi
+    fi
+else
```

Comment:
i sort of second that. if you have a block
```
if not condition
then 
   do_no
else
   do_yes
fi
```
you should always reorder it to
```
if condition
then
  do_yes
else
  do_no
fi
```
it's just more readable.